app/components/search_result/location_item_component.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
```

Comment:
This seems to be repeating a lot of what is in module AccessPanels.LocationItemComponent. Is there a reason we have to repeat all the same code. Could we use inheritance or something along those lines to not repeat code?

Reply:
Yes, but I believe the way to do that is by extracting a new class rather than forcing inheritance.  I'm trying not to make this PR too challenging to review, thus I've avoided this at this time.

Reply:
See https://github.com/sul-dlss/SearchWorks/pull/4151 as a way to preemptively reduce the duplication.